### PR TITLE
Accomodate Gio changes

### PR DESCRIPTION
**Pull request by Hofer-Julian**
- Fixes examples
- Fixes books
- Let the book tests also run when the book was not changed (don't think that fixes the problem we discussed though @bilelmoussaoui)

### CODE REVIEW

File: .github/workflows/book.yml [hunk 1]
```diff
@@ -5,8 +5,6 @@ on:
   push:
     branches:
       - "master"
-    paths:
```
Comment (sdroege) [inline .github/workflows/book.yml:8]: The changes to the book directory and this here should probably be in a separate commit from the "Fix examples" commit?

Comment (Hofer-Julian) [reply .github/workflows/book.yml:8]: Done

Comment (bilelmoussaoui) [reply .github/workflows/book.yml:8]: This should fix the issue yes, although, I wonder if there's a way to specify a whole directory with the files in it to trigger a build.

Comment (Hofer-Julian) [reply .github/workflows/book.yml:8]: In that case, that's not what we want though.
Changes outside the book can break the book examples as well.

Comment (bilelmoussaoui) [reply .github/workflows/book.yml:8]: They won't pass the CI though as the job is always run for every PR

Comment (Hofer-Julian) [reply .github/workflows/book.yml:8]: Ah, I see

**Comment by bilelmoussaoui** [comment]
I don't believe the CI changes were acked by me. Please don't merge directly next time.

**Comment by Hofer-Julian** [comment]
Sorry @bilelmoussaoui, I thought this PR already had one approval, but it indeed did not.

**Comment by sdroege** [comment]
Anything needs fixes here still?

**Comment by bilelmoussaoui** [comment]
Not really, it's just the CI "solution" doesn't seems right to me; I will take care of that along with #270

**Comment by Hofer-Julian** [comment]
Again, I said that I DON'T see that as the solution of the CI problem (see above 😉)

**Comment by bilelmoussaoui** [comment]
it's not a solution if you merge directly to master, which you should never do :-)
